src/base/bittorrent/session.h

```diff
@@ -565,14 +568,14 @@ namespace BitTorrent
         void saveTorrentsQueue();
         void removeTorrentsQueue();
 
-        void getPendingAlerts(std::vector<lt::alert *> &out, ulong time = 0);
+        std::vector<lt::alert *> getPendingAlerts(lt::time_duration time = lt::time_duration::zero()) const;
 
         // BitTorrent
-        lt::session *m_nativeSession;
+        lt::session *m_nativeSession = nullptr;
 
-        bool m_deferredConfigureScheduled;
-        bool m_IPFilteringChanged;
-        bool m_listenInterfaceChanged; // optimization
+        bool m_deferredConfigureScheduled = false;
+        bool m_IPFilteringChanged = true;
```

Comment:
Well, I see that it is because of code unification... But it still not obvious in this place why this values are initially "changed". I think there should be a comment on it.
Or we need to use some opposite variables (e.g. m_IPFilterConfigured and m_listenInterfaceConfigured).

Reply:
>I think there should be a comment on it.

Where should I put the comment?

>we need to use some opposite variables (e.g. m_IPFilterConfigured and m_listenInterfaceConfigured).

You mean replace `m_IPFilteringChanged` by `m_IPFilterConfigured`?

Reply:
>Where should I put the comment?

Comment should be placed near these initializers to explain why these variables are initially "true".

>You mean replace m_IPFilteringChanged by m_IPFilterConfigured?

Yes.
But not the names only. Since they will have opposite meaning all conditions with it should be changed as well (e.g. "if (m_IPFilterChanged)" => "if (!m_IPFilterConfigured)").

>Hmm, I don't prefer strongly of one over another

Since "just created thing is changed" is illogical for me I prefer the second way ("just created thing is not configured yet").